benchmarks/bench_ml.py

```diff
@@ -79,385 +70,336 @@ def pow2intmap(x):
 def nop(x):
     return x
 
-# this is used to process the output of fetch_mldata
-def load_data_target(data):
-    X = data['data']
-    Y = data['target']
-    return X, Y
-
-class MLBench():
+nnparams = {
+    # up to 1024 neurons
+    'hidden_layer_sizes': (Real(1.0, 10.0), pow2intmap),
+    'activation': (Categorical(['identity', 'logistic', 'tanh', 'relu']), nop),
+    'solver': (Categorical(['lbfgs', 'sgd', 'adam']), nop),
+    'alpha': (Real(-5.0, -1), pow10map),
+    'batch_size': (Real(5.0, 10.0), pow2intmap),
+    'learning_rate': (Categorical(['constant', 'invscaling', 'adaptive']), nop),
+    'max_iter': (Real(5.0, 8.0), pow2intmap),
+    'learning_rate_init': (Real(-5.0, -1), pow10map),
+    'power_t': (Real(0.01, 0.99), nop),
+    'momentum': (Real(0.1, 0.98), nop),
+    'nesterovs_momentum': (Categorical([True, False]), nop),
+    'beta_1': (Real(0.1, 0.98), nop),
+    'beta_2': (Real(0.1, 0.9999999), nop),
+}
+
+
+MODELS = {
+    MLPRegressor(): nnparams,
+    SVR(): {
+            'C': (Real(-4.0, 4.0), pow10map),
+            'epsilon': (Real(-4.0, 1.0), pow10map),
+            'gamma': (Real(-4.0, 1.0), pow10map)},
+    DecisionTreeRegressor(): {
+            'max_depth': (Real(1.0, 4.0), pow2intmap),
+            'min_samples_split': (Real(1.0, 8.0), pow2intmap)},
+    MLPClassifier(): nnparams,
+    SVC(): {
+            'C': (Real(-4.0, 4.0), pow10map),
+            'gamma': (Real(-4.0, 1.0), pow10map)},
+    DecisionTreeClassifier(): {
+            'max_depth': (Real(1.0, 4.0), pow2intmap),
+            'min_samples_split': (Real(1.0, 8.0), pow2intmap)}
+}
+
+# every dataset should have have a mapping to the mixin that can handle it.
+DATASETS = {
+    "Boston": RegressorMixin,
+    "Housing": RegressorMixin,
+    "digits": ClassifierMixin,
+    "Climate Model Crashes": ClassifierMixin,
+}
 
+# bunch of dataset preprocessing functions below
+def split_normalize(X, y):
```

Comment:
I removed the random state here because the black-box function is assumed to be noisy.